client/src/App/Header/UserSelect.tsx

```diff
@@ -17,7 +18,9 @@ const UserSelect: React.FC = () => {
     const open = Boolean(anchorEl);
 
     const { getCurrentUser, logout } = useAuthContext();
-    const currentUser = getCurrentUser();
+    const currentUser: UserInfo | null = useMemo(() => {
+        return getCurrentUser();
```

Comment:
you don't need return in arrow function. Also dependency array in useMemo works on values. Can we actually pass a function there?